src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpWinForms.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpWinForms : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpWinForms(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpWinForms), WellKnownProjectTemplates.WinFormsApplication, clearEditor: false)
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.WinForms)]
+        public void Add_Control()
+        {
+            AddWinFormButton(ProjectName, "Form1.cs", "SomeButton");
+            CloseFile(ProjectName, "Form1.cs", saveFile: true);
```

Comment:
Is it necessary to close the file before opening Form1.Designer.cs? Wouldn't a save be sufficient?